argocd/resource_argocd_application_test.go

```diff
@@ -424,6 +424,53 @@ func TestAccArgoCDApplication_Recurse(t *testing.T) {
 	})
 }
 
+func TestAccArgoCDApplication_Finalizers(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccArgoCDApplicationSimple(acctest.RandomWithPrefix("test-acc")),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckModuleResourceAttrNotSet(
+						"argocd_application.simple",
+						"metadata.0.finalizers",
+					),
+				),
+			},
+			{
+				Config: testAccArgoCDApplicationSimpleFinalizers(acctest.RandomWithPrefix("test-acc"), "[]"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"argocd_application.simple",
+						"metadata.0.finalizers.#",
+						"0",
+					),
+				),
+			},
+			{
+				Config: testAccArgoCDApplicationSimpleFinalizers(acctest.RandomWithPrefix("test-acc"), "[\"resources-finalizer.argocd.argoproj.io\"]"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"argocd_application.simple",
+						"metadata.0.finalizers.#",
+						"1",
+					),
+					resource.TestCheckResourceAttr(
+						"argocd_application.simple",
+						"metadata.0.finalizers.0",
+						"resources-finalizer.argocd.argoproj.io",
+					),
+				),
+			},
```

Comment:
Please add test step to verify importing resources as well. E.g.
https://github.com/oboukili/terraform-provider-argocd/blob/3531a32e53d7f12a5fd7f6a2868bc0ceda83fea1/argocd/resource_argocd_application_test.go#L45-L50